### PR TITLE
Create db migrations, models, model specs

### LINKED ISSUE #25: Create database migrations, models and model specs
[Database plan](https://docs.google.com/document/d/1NgCPNFg23u8QKuxuCy9QkiJ_typfUXG7MKMeApaCDHI/edit#)

### PR DESCRIPTION
**Pull request by leahriffell**
Dan + Leah Wrote Tests 
Dan + Leah Implemented
Dan + Leah Reviewed

Necessary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally
    
    [x] Ran and addressed RuboCop messages 

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes:

   closes #25 

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed (Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [x] I have commented on my PR, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link: https://giphy.com/gifs/mtvcribs-organization-kathy-griffin-organized-1oJRRiSteRVLHf7JKH

### CODE REVIEW

File: config/initializers/inflections.rb [hunk 1]
```diff
@@ -12,5 +12,5 @@
 
 # These inflection rules are supported but not enabled by default:
 # ActiveSupport::Inflector.inflections(:en) do |inflect|
-#   inflect.acronym 'RESTful'
+#   inflect.irregular 'party', 'parties'
```
Comment (leahriffell) [inline config/initializers/inflections.rb:15]: We thought that maybe an issue that we were running into was that party wasn't properly pluralizing, but it wasn't related so this code has been commented out. 

The issue was that our project is called viewing_party, so a module called ViewingParty was automatically created. This caused issues with our table name being called ViewParties. Resolved by renaming to Parties.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  has_many :friendships, dependent: :destroy
```
Comment (leahriffell) [inline app/models/user.rb:4]: RuboCop asserts that any relationship like this has a 'dependent' after it.

File: db/migrate/20201006213104_create_movies.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateMovies < ActiveRecord::Migration[5.2]
+  def change
+    create_table :movies do |t|
+
```
Comment (leahriffell) [inline db/migrate/20201006213104_create_movies.rb:4]: For now we created a Movies table w/ just ID. Once we have consumed the API, we will also add the API's movie ID.

File: db/migrate/20201006223108_create_party_users.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreatePartyUsers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :party_users do |t|
+      t.references :party, foreign_key: true
+      t.references :user, foreign_key: true
+      t.integer :attendee_type, default: 1
```
Comment (leahriffell) [inline db/migrate/20201006223108_create_party_users.rb:6]: PartyUser will default to being a guest (value 1) vs a host (value 0).

File: spec/models/friendship_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+describe Friendship, type: :model do
+  describe 'relationships' do
+
+    it { should belong_to :user }
+    it { should belong_to :friend }
+  end
+end
```
Comment (leahriffell) [inline spec/models/friendship_spec.rb:8]: We can build out this factory + others needed as the need for test objects arise.

File: app/controllers/welcome_controller.rb [hunk 6]
```diff
@@ -1,4 +1,3 @@
-class WelcomeController < ApplicationController 
-  def index 
-  end
-end
\ No newline at end of file
+class WelcomeController < ApplicationController
+  def index; end
```
Comment (dhalverson) [inline app/controllers/welcome_controller.rb:2]: This is interesting, will look into this more later

**Review by dhalverson — APPROVED** [review]
Looks good

**Pull request merged**